### ISSUE TITLE
Figure out how to get language data

### ISSUE THREAD

**Issue by tfmorris** [Git-Lit/git-lit#24]
The metadata METS/MODS files only have a language declared for a tiny fraction of the files.  The `booklist.tsv` file has a language for all the files (although it's `und` for 19K), but it doesn't appear to have an identifier which is present in the _metadata.xml files.

We need a way to join both sets of info to provide a complete set of metadata (or perhaps derive language data empirically).

**Comment by JonathanReeve** [comment]
This looks promising: https://pypi.python.org/pypi/langdetect. It'd be fun to play around with sometime.

**Comment by tfmorris** [comment]
It'd be better to get it from the metadata, if we could, but if we have fall back to language detection from text strings, CLD2 & language-detection are the two leading candidates (although the referenced Python wrapper for language-detection points to an obsolete site, which isn't inspiring).

http://blog.mikemccandless.com/2013/08/a-new-version-of-compact-language.html
https://github.com/CLD2Owners/cld2
https://github.com/shuyo/language-detection
https://pypi.python.org/pypi/cld2-cffi

**Comment by tfmorris** [comment]
We didn't actually have the data necessary to link everything together, so I had to resort to scraping it from the British Library catalog.  They actually create a separate catalog record, with a different system number (sysnum), for the scanned digital resource.  It's this sysnum which is booklist.tsv, but all the other files use the original sysnum.

The mapping between the two sets of numbers is in https://github.com/Git-Lit/git-lit/blob/master/metadata/crosswalk.tsv and I've used it to create `booklist-extended.tsv` which has the print sysnum and a computed language added to it.  The language detection is done using both CLD2 and language-detection on a cleaned up version of the title and only used if both agree.

In addition to providing a reasonable language guess for 15,000 of the titles that weren't coded, it also identified a couple hundred which appear to be coded wrong.  This list includes a few false positives, but is mostly made up of records which really do need to be fixed.

The last column in `booklist.tsv` is a list of identifiers which point to the online PDF and JPEG2000 viewer scanned volumes.  We can use these to link to the page images.

As expected, the unknown language records are overwhelming English, like the rest of the collection.  A quick glance at the 3,000 editions which still don't have a language coded confirms that the bulk of them are English, too, so the final tally will be ~37,000 English editions out of the ~50,000. A rough breakdown of the top languages:

![image](https://cloud.githubusercontent.com/assets/82178/12661181/2261f5b6-c5e6-11e5-9923-a1d4b2d6b62e.png)
